wpilibj/src/main/java/edu/wpi/first/wpilibj/GenericHID.java

```diff
@@ -397,16 +399,24 @@ public void setOutputs(int value) {
    * @param value The normalized value (0 to 1) to set the rumble to
    */
   public void setRumble(RumbleType type, double value) {
-    if (value < 0) {
-      value = 0;
-    } else if (value > 1) {
-      value = 1;
+    value = MathUtil.clamp(value, 0, 1);
+    short rumbleValue = (short) (value * 65535);
```

Comment:
A short in Java only goes up to 32767.  This looks like it may have been a bug in the original implementation too?

Reply:
im not sure, this feature goes back a long time in the git history. I have used this method before on a robot without it crashing and rumbling just fine. Im not sure if the JNI native code does something to check if a -1 value is 100% output. I can also just change this to 32767 instead of 65535.